pkg/kubectl/service.go

```diff
@@ -183,8 +183,16 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 	if len(params["type"]) != 0 {
 		service.Spec.Type = api.ServiceType(params["type"])
 	}
-	if service.Spec.Type == api.ServiceTypeLoadBalancer {
+	switch service.Spec.Type {
+	case api.ServiceTypeNodePort:
+		for i, port := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = port.Port
+		}
+	case api.ServiceTypeLoadBalancer:
 		service.Spec.LoadBalancerIP = params["load-balancer-ip"]
+	case "Headless":
```

Comment:
I would prefer --clusterIP=None for now, and describe headless in the
expose doc.

On Mon, May 9, 2016 at 11:47 PM, Tim Hockin <notifications@github.com>
wrote:

> In pkg/kubectl/service.go
> <https://github.com/kubernetes/kubernetes/pull/25034#discussion_r62608700>
> :
>
> >  		service.Spec.LoadBalancerIP = params["load-balancer-ip"]
> > +	case "Headless":
>
> While type==Headless seems semantically sane, I think it will encourage
> people to try that in the YAML/JSON (regardless of your warnings in the
> docs). I think we should just say --clusterIP=None creates a headless
> Service, or else we should actually add support for type=Headless.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25034/files/fde8d6248f144aa8723e6d46946ffb595de888ad#r62608700>
>